readinglist/__init__.py

```diff
@@ -97,9 +108,36 @@ def on_new_response(event):
     config.add_subscriber(on_new_response, NewResponse)
 
 
+def end_of_life_tween_factory(handler, registry):
+    """Pyramid tween to handle service end of life."""
+
+    def eos_tween(request):
+        eos_date = registry.settings.get("readinglist.eos")
+        eos_url = registry.settings.get("readinglist.eos_url")
+        if eos_date:
+            eos_date = dateparser.parse(eos_date)
```

Comment:
Ah, I see :(

Alternatives are:
* create a new app with some settings that have eol enabled
* unit test of this factory and simply test that tween is registered
* use a "cache" in the closure to do it once only
* keep it this way :)